### DISCUSSION TITLE
BrightScript inside CDATA within XML file when using VSCode causing issues

### DISCUSSION THREAD

**Discussion by kosso** [rokucommunity/vscode-brightscript-language#569] [Q&A] [answered]
Hi,
I'm extremely new to Roku development, but I feel like I'm getting somewhere after going through the many samples out there. (Expect many more questions from me. :) - I have already implemented our music videos app on iOS, Android, Amazon FireTV, AndroidTV, AppleTV, webOS and Tizen)

I'm having an issue using the BRS VSCode extension where if I open an XML file which contains BrightScript inside CDATA, the `interface` `field` elements, which have an `onChange` reference are unable to recognise the functions in the script tag below.  "Cannot find [function name] in component scope"

If I move the BS from the script tag in the XML to its own .brs file it works. 

Is there some setting to make this work? Or is this anything to do with legacy samples using old methods? 

Thanks!

**Comment by TwitchBronBron** [comment, marked as answer]
Welcome to Roku Development, and thanks for using the RokuCommunity tools! If you haven't yet, you should also join us on [slack](https://join.slack.com/t/rokudevelopers/shared_invite/zt-4vw7rg6v-NH46oY7hTktpRIBM_zGvwA). 

Unfortunately there's no setting to enable support for CDATA in xml files. While it's technically valid syntax, it's widely considered bad coding style across the BrightScript development community, and almost every dev shop I'm aware of actively prohibits that coding style in their projects. That makes it a very low priority for us at RokuCommunity to add support since almost nobody allows it in their projects anyway, and we have many higher priority features and fixes to implement.

Unfortunately many of the Roku sample projects still have this CDATA style coding so it does make things a little confusing when you're first getting started.

**Reply by kosso** [reply]
Ah OK. I suspected that  might be the case :) 
I can split them up for testing these samples.  

Thanks for getting back to me so quickly.

The VSCode tools have proven to be invaluable for development and learning. (Steep learning curve though)
Before finding them, I found myself in the process of trying to build a Chrome Puppeteer script to automate the file watching, zipping and file upload!  :D
